Editor/Scripts/BotManagement/RGBotSynchronizer.cs

```diff
@@ -363,26 +410,33 @@ private void CreateParentFolders()
         private string CreateBotFolder(string botName, long botId)
         {
             CreateParentFolders();
-            var folderString = $"{BOTS_PATH}/{botName}_{botId}".Replace('-','n');
-            AssetDatabase.CreateFolder(BOTS_PATH, $"{botName}_{botId}".Replace('-','n'));
+            var botFolderName = $"{botName}_{botId}".Replace('-', 'n');
+            var folderString = $"{BOTS_PATH}/{botFolderName}";
+            if (AssetDatabase.IsValidFolder(folderString))
```

Comment:
I should have put a code comment here I guess when I first wrote this, but this auto delete code was intentionally excluded.

If they have a directory that matches this new bot being synced, but that directory does not have a bot asset record in it that would have allowed us to do the proper checks...  I chose not to wipe out whatever they had in that directory blindly and instead let the process throw an error so they could resolve the situation themselves and save whatever was in that directory from getting deleted.

Reply:
Hmm, there's probably more we have to do to handle that scenario, since `AssetDatabase.CreateFolder` won't error out if the directory does not exist. Instead, it [appends a sequence number to the end](https://docs.unity3d.com/ScriptReference/AssetDatabase.CreateFolder.html). That does preserve the existing code, but our logic for identifying Bot code on disk is highly dependent upon the path so I don't think the new bot will be properly synced with the server. It might be best to just throw an error in that case, but I'm happy to do a little bit of quick testing and file a separate ticket for handling that.

For now, I'll have a `replace` argument that flows in here from RGBotSynchronizer to make it more explicit when we **want** auto-deleting here. We _do_ want it when we're synchronizing something we know to be a remote bot, otherwise old files don't get cleaned up.

Reply:
Yeah, it looks like the existing behavior for conflicting directories is a little unexpected, but that's a separate issue. I filed [REG-1430](https://linear.app/regression-games/issue/REG-1430/unity-bot-sync-behavior-is-unexpected-if-directory-exists-and-is-not-a) from my investigation.

For this PR, I'll leave that part of the behavior and make auto-delete conditional on the `replace` parameter, as I've done in the latest iteration. That will preserve the current behavior for directories that already exist but aren't bots.